### PR TITLE
Bug 1958913: discover-etcd-initial-cluster: retry if member is not part of member list and dataDir exists

### PR DESCRIPTION
**Pull request by hexfusion**
Signed-off-by: Sam Batschelet <sbatsche@redhat.com>

Please read https://github.com/etcd-io/etcd/blob/master/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go [hunk 1]
```diff
@@ -198,15 +198,15 @@ func (o *DiscoverEtcdInitialClusterOptions) Run() error {
...
 		// Condition: member not found with dataDir
-		// The member has been removed from the cluster. etcd should not start and return error. The dataDir will be archived once the operator
+		// The member has been removed from the cluster. The dataDir will be archived once the operator
 		// scales up etcd.
-		// Result: return error and restart container
+		// Result: retry member check allowing operator time to scale up etcd again on this node.
 		if !found && dataDirExists {
-			return fmt.Errorf("member %q is no longer a member of the cluster and should not start", target.String())
+			fmt.Fprintf(os.Stderr, "      member %q not found in member list but dataDir exists, check operator logs for possible scaling problems\n", target.String())
 		}
```
Comment (retroflexer) [inline openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go:211]: Before your recent fix, we used to return with success if dataDir was present:
https://github.com/openshift/etcd/commit/af729a8c3a354aa7eee3e27eb10658173fda613d#diff-3d1414e0d47047e4fdcae958fb7654d082c1eb1b71cf4eb38aeb3190db678208L168-R172

**Comment by hexfusion** [comment]
unit tests need to be reworked I believe.

**Comment by retroflexer** [comment]
> unit tests need to be reworked I believe.

Yeah, the spelling for the destroyed :laughing: 
```
"member \"unixs://127.0.0.1:2100120481\" dataDir has been destoyed and must be removed from the cluster"  (expected)
"member \"unixs://127.0.0.1:2100120481\" dataDir has been destroyed and must be removed from the cluster" (got)
```

File: openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go (hunk 1 above)
Comment (hexfusion) [reply openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go:211]: Yeah in the eyes of the operator this is essentially scaling up workflow so it feels reasonable that we give the container a bit longer to be observable.

**Comment by hexfusion** [comment]
/hold QE is in the process of testing for verification. 

@geliu2016 unhold if that testing works as expected.

**Comment by hexfusion** [comment]
testing was verified by QE: https://bugzilla.redhat.com/show_bug.cgi?id=1958913

/hold cancel.

**Pull request merged**